pkg/tsdb/cloudwatch/types.go

```diff
@@ -54,3 +54,17 @@ const (
 	GMDApiModeMathExpression
 	GMDApiModeSQLExpression
 )
+
+type fillMissing struct {
+	mode  fillMode
+	value float64
+}
+
+type fillMode int
+
+const (
+	null     fillMode = iota // fills with null, the default behavior
+	previous                 // fills with the last seen value unless that does not exist, in which case it fills with null
+	value                    // fills with a specific value
+	optOut                   // does not fill
```

Comment:
You can get the existing fill modes from the grafana-plugin-sdk. There is the package `data` there that defines `FillMode` for previous, null and value. My suggestion is to do "opt out" if the value set there is not any of the other 3 (e.g. `4`).

We can try and submit a suggestion for a fourth value in the `FillMode`, something like `FillModeSkip` so other plugins can follow this approach (not needed for this PR).

Reply:
An advantage to our own values is simply that we start our default `null` at 0 so that even without defining the `fillMissing` in cloudwatch, we are always guaranteed that default behavior. We'll never use the sdk's 0 (=previous value) by accident.

In other words, is there any other way to be quite careful so that `null` is our default?
I could revert to using a pointer for the `fillMissing` struct; that would introduce checks for nil everywhere and manually initializing the values to sdk's 1 (=null). 
It seems less clear and concise. 

Let me know what you think.

Reply:
> An advantage to our own values is simply that we start our default null at 0 so that even without defining the fillMissing in cloudwatch, we are always guaranteed that default behavior. We'll never use the sdk's 0 (=previous value) by accident.

yeah, that is also a disadvantage. It's not ideal that fillMode: 0 means different things for different plugins (one for cloudwatch and other for all the other plugins). The goal would be to standardize things as much as possible.

> In other words, is there any other way to be quite careful so that null is our default?

This should be easy to unit-test so we can ensure that we are not changing other default value by mistake.

> I could revert to using a pointer for the fillMissing struct; that would introduce checks for nil everywhere and manually initializing the values to sdk's 1 (=null).

Not needed, I think it's fine not using a pointer for this.